apps/condo/domains/ticket/hooks/useTicketDeferModal.tsx

```diff
@@ -0,0 +1,140 @@
+    const InfoBlockContentMessage = intl.formatMessage({ id: 'pages.condo.ticket.id.TicketDefer.DeferDateModal.InfoBlock.content' })
+    const SaveButtonLabelMessage = intl.formatMessage({ id: 'pages.condo.ticket.id.TicketDefer.DeferDateModal.SaveButton.label' })
+    const CancelModalTitleMessage = intl.formatMessage({ id: 'pages.condo.ticket.id.TicketDefer.CancelModal.title' })
+    const CancelModalContentMessage = intl.formatMessage({ id: 'pages.condo.ticket.id.TicketDefer.CancelModal.content' })
+    const SelectDateButtonLabelMessage = intl.formatMessage({ id: 'pages.condo.ticket.id.TicketDefer.CancelModal.SelectDateButton.label' })
+    const LeaveCurrentStatusButtonLabelMessage = intl.formatMessage({ id: 'pages.condo.ticket.id.TicketDefer.CancelModal.LeaveCurrentStatusButton.label' })
+
+    const [isDeferModalVisible, setIsDeferModalVisible] = useState<boolean>(false)
+    const [isCancelModalVisible, setIsCancelModalVisible] = useState<boolean>(false)
+    const [deferredDate, setDeferredDate] = useState<Dayjs>(null)
+    const [statusDeferredId, setStatusDeferredId] = useState<string | null>(null)
+
+    const openDeferDateModal = useCallback(() => setIsDeferModalVisible(true), [])
+
+    const closeDeferDateModal = useCallback(() => setIsDeferModalVisible(false), [])
+
+    const openCancelModal = useCallback(() => setIsCancelModalVisible(true), [])
+
+    const closeCancelModal = useCallback(() => setIsCancelModalVisible(false), [])
+
+    const handleOpenDeferDateModal = useCallback((statusDeferredId: string) => {
+        setStatusDeferredId(statusDeferredId)
+        openDeferDateModal()
+    }, [openDeferDateModal])
+
+    const handleReset = useCallback(() => {
+        setDeferredDate(null)
+        closeDeferDateModal()
+        closeCancelModal()
+        setStatusDeferredId(null)
+    }, [closeDeferDateModal, closeCancelModal])
+
+    const handleCloseDeferTicketModal = useCallback(() => {
+        setDeferredDate(null)
+        closeDeferDateModal()
+        openCancelModal()
+    }, [closeDeferDateModal, openCancelModal])
+
+    const handleSaveDeferredTicket = useCallback(() => {
+        statusDeferredId && updateTicket(statusDeferredId, deferredDate)
+        handleReset()
+    }, [deferredDate, handleReset, statusDeferredId, updateTicket])
+
+    const handleOpenDeferTicketModal = useCallback(() => {
+        closeCancelModal()
+        openDeferDateModal()
+    }, [closeCancelModal, openDeferDateModal])
+
+    const handleDisabledDate = useCallback((current) => current < minDate || current > maxDate, [])
+
+    const deferDateModal = useMemo(() => (
+        <Modal
+            visible={isDeferModalVisible}
+            onCancel={handleCloseDeferTicketModal}
+            title={DeferDateModalTitleMessage}
+            footer={
+                <Button
+                    onClick={handleSaveDeferredTicket}
+                    disabled={!deferredDate}
+                    type='sberDefaultGradient'
```

Comment:
`'` is for js strings
`"` is for HTML attributes
```suggestion
                    type="sberDefaultGradient"
```

Reply:
single quotes are used everywhere in the project, and in such cases, too

Reply:
why should double quotes be used if single quotes are everywhere?